app/services/foreman_register/registration_url.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ForemanRegister
+  class RegistrationUrl
+    attr_reader :host
+
+    delegate :registration_token, to: :host
+
+    def initialize(host:)
+      @host = host
+    end
+
+    def url
+      uri = URI.parse(Setting[:foreman_url])
+      uri.scheme = SETTINGS[:require_ssl] ? 'https' : 'http'
+      uri.path = Rails.application.routes.url_helpers.register_foreman_register_hosts_path
+      uri.query = "token=#{registration_token}"
```

Comment:
The API docs are not very clear on whether you need to encode this or if the URI class does this for you. Do you know?

Reply:
Hm. Does it matter? I believe the token should always be URL safe.

Reply:
Hmm, probably ok. I've just gotten used to playing it safe and passing hashes to such libraries which then handle it safely for me so I don't need to think about it. When I pass in strings, I always question it.